fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/block/type/BlockSetTypeBuilder.java

```diff
@@ -0,0 +1,255 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.object.builder.v1.block.type;
+
+import net.minecraft.block.BlockSetType;
+import net.minecraft.sound.BlockSoundGroup;
+import net.minecraft.sound.SoundEvent;
+import net.minecraft.sound.SoundEvents;
+import net.minecraft.util.Identifier;
+
+/**
+ * This class allows easy creation of {@link BlockSetType}s.
+ *
+ * <p>A {@link BlockSetType} is used to tell the game various properties of related blocks, such as what sounds they should use.
+ *
+ * @see WoodTypeBuilder
+ */
+public final class BlockSetTypeBuilder {
+	private boolean openableByHand = true;
+	private BlockSoundGroup soundGroup = BlockSoundGroup.WOOD;
+	private SoundEvent doorCloseSound = SoundEvents.BLOCK_WOODEN_DOOR_CLOSE;
+	private SoundEvent doorOpenSound = SoundEvents.BLOCK_WOODEN_DOOR_OPEN;
+	private SoundEvent trapdoorCloseSound = SoundEvents.BLOCK_WOODEN_TRAPDOOR_CLOSE;
+	private SoundEvent trapdoorOpenSound = SoundEvents.BLOCK_WOODEN_TRAPDOOR_OPEN;
+	private SoundEvent pressurePlateClickOffSound = SoundEvents.BLOCK_WOODEN_PRESSURE_PLATE_CLICK_OFF;
+	private SoundEvent pressurePlateClickOnSound = SoundEvents.BLOCK_WOODEN_PRESSURE_PLATE_CLICK_ON;
+	private SoundEvent buttonClickOffSound = SoundEvents.BLOCK_WOODEN_BUTTON_CLICK_OFF;
+	private SoundEvent buttonClickOnSound = SoundEvents.BLOCK_WOODEN_BUTTON_CLICK_ON;
+
+	/**
+	 * Sets whether this block set type's door and trapdoor can be opened by hand.
+	 *
+	 * <p>Defaults to {@code true}.
+	 *
+	 * @return this builder for chaining
+	 */
+	public BlockSetTypeBuilder openableByHand(boolean openableByHand) {
+		this.openableByHand = openableByHand;
+		return this;
+	}
+
+	/**
+	 * Sets this block set type's door and trapdoor to be openable by hand.
+	 *
+	 * @return this builder for chaining
+	 */
+	public BlockSetTypeBuilder openableByHand() {
+		this.openableByHand = true;
+		return this;
+	}
+
+	/**
+	 * Sets this block set type's door and trapdoor to be unopenable by hand.
+	 *
+	 * @return this builder for chaining
+	 */
+	public BlockSetTypeBuilder unopenableByHand() {
+		this.openableByHand = false;
+		return this;
+	}
```

Comment:
Not too sure about these two, especially the last, but I guess they don't hurt.

Reply:
I agree, I think generally id just go with the `openableByHand(boolean openableByHand)` 👍

Reply:
Removed these shortcut methods in latest commit